src/uwtools/tests/utils/test_api.py

```diff
@@ -118,20 +118,19 @@ def test_str2path_convert():
     assert result == Path(val)
 
 
-def test__execute(execute_kwargs, tmp_path):
-    config = tmp_path / "config.yaml"
-    with open(config, "w", encoding="utf-8") as f:
-        yaml.dump({"some": "config"}, f)
+@pytest.mark.parametrize("hours", [0, 24, 168])
+def test__execute(execute_kwargs, hours, tmp_path):
     graph_file = tmp_path / "g.dot"
-    kwargs = {
-        **execute_kwargs,
-        "driver_class": ConcreteDriver,
-        "config": config,
-        "cycle": dt.datetime.now(),
-        "leadtime": dt.timedelta(hours=24),
-        "graph_file": graph_file,
-    }
-    assert not graph_file.is_file()
-    with patch.object(ConcreteDriver, "_validate"):
+    with patch.object(test_driver, "ConcreteDriver", wraps=test_driver.ConcreteDriver) as cd:
+        kwargs = {
+            **execute_kwargs,
+            "driver_class": cd,
+            "config": {"some": "config"},
+            "cycle": dt.datetime.now(),
+            "leadtime": dt.timedelta(hours=hours),
+            "graph_file": graph_file,
+        }
+        assert not graph_file.is_file()
         assert api._execute(**kwargs) is True
+        assert cd.call_args.kwargs["leadtime"] == dt.timedelta(hours=hours)
```

Comment:
Here's the assertion that passes with the fix on the parent PR and fails without it: The driver class' constructor will be called with a `timedelta` argument for `leadtime` if and only if a `leadtime` -- including a `0` one, with the parent-PR fix -- is provided to the API.